dependencies/editline/meson.build

```diff
@@ -0,0 +1,33 @@
+# Editline Dependency File
+#============================================================================
+
+
+# Look for editline, a required dependency.
+#--------------------------------------------------
+# NOTE: The the libeditline.pc file was added only in libeditline >= 1.15.2, see 
+# https://github.com/troglobit/editline/commit/0a8f2ef4203c3a4a4726b9dd1336869cd0da8607,
+# but e.g. Ubuntu 16.04 has an older version, so we fall back to searching for
+# editline.h when the pkg-config approach fails.
+
+editline_dep = declare_dependency(
+    dependencies : cpp.find_library('editline'),
+    link_args :  get_option('editline_link_args'))
+
+
+if not (
+    cpp.has_function(
+        'read_history',
+        prefix : '#include <stdio.h>\n#include "editline.h"',
+        args : '-lreadline'))
+    warning('Nix requires libeditline; However, required functions do not work. Maybe ' +\
```

Comment:
If the required functions does not work, why is this a mere warning and not a error?

Reply:
i set it to a warning whilst writing it and forgot to change it to an error :*